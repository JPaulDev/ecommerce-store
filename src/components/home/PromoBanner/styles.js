import styled from 'styled-components';

export const Container = styled.div`
  background-color: #ebebeb;
  background-image: url(${({ background }) => background});
  background-position: calc(100% + 220px) 50%;
  background-repeat: no-repeat;
  background-size: auto 186px;
  border: 1px solid var(--color-grey-12);
  margin-top: 15px;
  min-height: 185px;
  overflow: hidden;
  padding: 15px;
  position: relative;

  @media screen and (min-width: 550px) {
    padding: 15px 25px;
  }

  @media screen and (min-width: 750px) {
    background-position: calc(100% + 345px) 50%;
    background-size: auto 291px;
    min-height: 250px;
    padding: 25px;
  }

  @media screen and (min-width: 850px) {
    background-position: center;
  }
`;

export const Heading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-28);

  @media screen and (min-width: 750px) {
    font-size: var(--font-size-38);
  }
`;

export const Text = styled.p`
  font-family: 'Uni Sans Book', sans-serif;
  line-height: 1.3;
  margin: 4px 0 10px;
  width: 45%;

  @media screen and (min-width: 750px) {
    font-size: var(--font-size-20);
    width: 35%;
  }
`;

export const LogoWrapper = styled.div`
  position: absolute;
  right: 25px;
  top: 25px;

  @media screen and (max-width: 849px) {
    display: none;
  }
`;

export const Wrapper = styled.div`
  bottom: -110px;
  position: absolute;
  right: -70px;

  @media screen and (min-width: 750px) {
    right: 10px;
  }
`;
