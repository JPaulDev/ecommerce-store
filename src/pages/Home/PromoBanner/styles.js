import styled from 'styled-components';

export const Container = styled.div`
  background-color: #ebebeb;
  background-image: url(${({ background }) => background});
  background-position: calc(100% + 220px) 50%;
  background-repeat: no-repeat;
  background-size: auto 186px;
  border: 1px solid var(--color-grey-9);
  margin-top: 15px;
  min-height: 185px;
  overflow: hidden;
  padding: 15px;
  position: relative;

  @media screen and (min-width: 550px) {
    padding: 15px 25px;
  }

  @media screen and (min-width: 750px) {
    background-position: calc(100% + 345px) 50%;
    background-size: auto 291px;
    min-height: 250px;
    padding: 25px;
  }

  @media screen and (min-width: 850px) {
    background-position: center;
  }
`;

export const Title = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.75rem;

  @media screen and (min-width: 750px) {
    font-size: 2.6rem;
  }
`;

export const Text = styled.p`
  font-family: 'Uni Sans Book', sans-serif;
  font-size: 1.06rem;
  line-height: 1.3;
  margin: 4px 0 10px;
  width: 45%;

  @media screen and (min-width: 750px) {
    font-size: 1.2rem;
    width: 35%;
  }
`;

export const Logo = styled.img`
  height: 73px;
  position: absolute;
  right: 25px;
  top: 25px;
  width: 93px;
`;

export const BlueDetail = styled.img`
  bottom: -105px;
  position: absolute;
  right: -70px;

  @media screen and (min-width: 750px) {
    bottom: -100px;
    right: 10px;
  }
`;
