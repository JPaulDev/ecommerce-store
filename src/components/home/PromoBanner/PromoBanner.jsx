import Image from 'next/future/image';
import nvidiaImage from '../../../../public/images/promo-banner/rtx-studio.webp';
import nvidiaLogo from '../../../../public/images/promo-banner/nvidia-logo.webp';
import blueDetail from '../../../../public/images/promo-banner/blue-detail.webp';
import useMediaQuery from '../../../hooks/useMediaQuery';
import { LinkButton, LinkChevron } from '../../common';
import * as Styled from './styles';

export default function PromoBanner() {
  const isMatch = useMediaQuery('(min-width: 530px)');
  const linkText = 'Find Out More';

  return (
    <Styled.Container background={nvidiaImage.src}>
      <Styled.Title>RTX Studio</Styled.Title>
      <Styled.Text>
        The best laptops and desktops for creating anywhere.
      </Styled.Text>
      {isMatch ? (
        <LinkButton href="/" linkText={linkText} padding="12px 14px" />
      ) : (
        <LinkChevron
          href="/"
          linkText={linkText}
          fontSize="0.8rem"
          iconSize={18}
        />
      )}
      <Styled.LogoWrapper>
        <Image src={nvidiaLogo} alt="" />
      </Styled.LogoWrapper>
      <Styled.Wrapper>
        <Image src={blueDetail} alt="" />
      </Styled.Wrapper>
    </Styled.Container>
  );
}
