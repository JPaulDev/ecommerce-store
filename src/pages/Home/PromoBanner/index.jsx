import useMediaQuery from '../../../hooks/useMediaQuery';
import nvidiaImage from '../../../assets/images/promo-banner/rtx-studio.webp';
import nvidiaLogo from '../../../assets/images/promo-banner/nvidia-logo.webp';
import blueDetail from '../../../assets/images/promo-banner/blue-detail.webp';
import LinkButtonGrey from '../../../components/LinkButtonGrey';
import LinkWithChevron from '../../../components/LinkWithChevron';
import * as Styled from './styles';

export default function AdvertBanner() {
  const isMatch530 = useMediaQuery('(min-width: 530px)');
  const isMatch850 = useMediaQuery('(min-width: 850px)');
  const linkText = 'Find Out More';

  return (
    <Styled.Container background={nvidiaImage}>
      <Styled.Title>RTX Studio</Styled.Title>
      <Styled.Text>
        The best laptops and desktops for creating anywhere.
      </Styled.Text>
      {isMatch530 ? (
        <LinkButtonGrey href="" padding="12px 14px">
          {linkText}
        </LinkButtonGrey>
      ) : (
        <LinkWithChevron
          url=""
          text={linkText}
          fontSize="0.75rem"
          iconSize="18px"
        />
      )}
      {isMatch850 && <Styled.Logo src={nvidiaLogo} alt="" />}
      <Styled.BlueDetail src={blueDetail} alt="" />
    </Styled.Container>
  );
}
