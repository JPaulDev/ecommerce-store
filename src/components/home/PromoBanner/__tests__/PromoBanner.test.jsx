import renderer from 'react-test-renderer';
import PromoBanner from '../index';

describe('PromoBanner component', () => {
  it('renders a promotional banner about RTX studio', () => {
    const tree = renderer.create(<PromoBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
