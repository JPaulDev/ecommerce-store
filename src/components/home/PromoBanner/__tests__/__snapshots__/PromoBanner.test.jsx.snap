// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoBanner component renders a promotional banner about RTX studio 1`] = `
<div
  className="sc-39860b04-0 ehules"
>
  <h4
    className="sc-39860b04-1 dMJGMK"
  >
    RTX Studio
  </h4>
  <p
    className="sc-39860b04-2 cwUlcb"
  >
    The best laptops and desktops for creating anywhere.
  </p>
  <a
    className="sc-bb098475-0 irrRtc"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Find Out More
    <svg
      fill="currentColor"
      height={18}
      viewBox="0 0 512 512"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
      />
    </svg>
  </a>
  <div
    className="sc-39860b04-3 RzafA"
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
  <div
    className="sc-39860b04-4 dvTVcY"
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
</div>
`;
